.ci/pitest.sh

```diff
@@ -18,12 +18,15 @@ case $1 in
   PROFILES=$(list_profiles "${0}");
 
   if [[ $(echo "$PROFILES" | grep -w -- "${1}" | cat) != "" ]]; then
-    mvn -e --no-transfer-progress -P"$1" clean test-compile org.pitest:pitest-maven:mutationCoverage
+    RESULT=0
+    mvn -e --no-transfer-progress -P"$1" clean test-compile \
+      org.pitest:pitest-maven:mutationCoverage | RESULT=1
```

Comment:
I don't understand why this drops the maven failure.

https://github.com/checkstyle/checkstyle/actions/runs/3926061139/jobs/6711436264
> SC2030 (info): Modification of RESULT is local (to subshell caused by pipeline).
> SC2031 (info): RESULT was modified in a subshell. That change might be lost.
I also don't know what to do to please shellcheck.

Reply:
Why not just `set +e` before and set it back to `set -e` . If the maven command execution fails for XYZ  reason, then the pit report won't be created (most likely) and the groovy command will subsequently fail.

Reply:
better to play with "set + e", we have examples in our scripts

Reply:
@Vyom-Yadav If you wish to take on this issue, please do. I am not familar enough with bash to resolve this.

Reply:
Created #12686, you can close this.